spec/ssl/gen_certs.sh

```diff
@@ -10,6 +10,10 @@ default_startdate = 2015010360000Z
 [ req ]
 distinguished_name = req_distinguished_name
 
+# Root CA certificate extensions
+[ v3_ca ]
+basicConstraints = critical, CA:true
```

Comment:
In my testing neither critical or pathlen attributes were strictly needed. We might not even need critical either for tests reading this SO post: https://security.stackexchange.com/questions/30974/which-properties-of-a-x-509-certificate-should-be-critical-and-which-not

Thoughts?